sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -449,8 +469,9 @@ __esimd_scatter_scaled(__ESIMD_DNS::simd_mask_storage_t<N> pred,
 
     for (int idx = 0; idx < N; idx++) {
       if (pred[idx]) {
-        Ty *addr = reinterpret_cast<Ty *>(elem_offsets[idx] + writeBase);
-        *addr = vals[idx];
+        RestoredTy *addr =
```

Comment:
For later: please check with CM library devs that acquiring a mutex is really needed when accessing through a cm buffer. That seems really strange requirement. It is also strange that no mutex is needed when accessing SLM.

Reply:
Will check with CM Lib devs.

Reply:
Issue ticket : https://github.com/intel/llvm/issues/5863